eng/validation/build-arcadewithrepo.ps1

```diff
@@ -0,0 +1,265 @@
+Param(
+  [Parameter(Mandatory=$true)]
+  [string] $azdoOrg, 
+
+  [Parameter(Mandatory=$true)]
+  [string] $azdoProject,
+
+  [Parameter(Mandatory=$true)]
+  [int] $buildDefinitionId,
+
+  [Parameter(Mandatory=$true)]
+  [string] $azdoToken,
+
+  [Parameter(Mandatory=$true)]
+  [string] $githubUser,
+  
+  [Parameter(Mandatory=$true)]
+  [string] $githubPAT,
+
+  [Parameter(Mandatory=$true)]
+  [string] $githubOrg, 
+
+  [Parameter(Mandatory=$true)]
+  [string] $targetRepoName,
+
+  [Parameter(Mandatory=$true)]
+  [string] $barToken, 
+
+  [string] $buildParameters = ''
+)
+
+set-strictmode -version 2.0
+$ErrorActionPreference = 'Stop'
+
+. $PSScriptRoot\..\common\tools.ps1
+. $PSScriptRoot\..\common\darc-init.ps1
+
+$arcadeSdkPackageName = 'Microsoft.DotNet.Arcade.Sdk'
+$arcadeSdkVersion = $GlobalJson.'msbuild-sdks'.$arcadeSdkPackageName
+
+# Get a temporary directory for a test root. Use the agent work folder if running under azdo, use the temp path if not.
+$testRootBase = if ($env:AGENT_WORKFOLDER) { $env:AGENT_WORKFOLDER } else { $([System.IO.Path]::GetTempPath()) }
+$testRoot = Join-Path -Path $testRootBase -ChildPath $([System.IO.Path]::GetRandomFileName())
+New-Item -Path $testRoot -ItemType Directory | Out-Null
+
+$days = 3
+$minTime = (Get-Date).AddDays(-$days)
+$buildReasonsList = @("batchedCI", "individualCI")
+
+function Get-LastKnownGoodBuildSha(
+    [string] $azdoOrg, 
+    [string] $azdoProject,
+    [int] $buildDefinitionId,
+    [string] $azdoToken)
+{
+
+    ## Have there been any builds in the last $days days? 
```

Comment:
Should be able to extract this into a Set-Headers function or something. This is duplicated in the Invoke function below.

Reply:
Get-Headers all the things :)